db/migrate/20221202130039_fill_project_journals_with_existing_data.rb

```diff
@@ -0,0 +1,95 @@
+class FillProjectJournalsWithExistingData < ActiveRecord::Migration[7.0]
+  def up
+    create_journal_entries_for_projects
+  end
+
+  def down
+    delete_journal_entries_for_projects
+  end
+
+  def create_journal_entries_for_projects
+    sql = <<~SQL.squish
+      WITH project_journals_insertion AS (
+        INSERT INTO project_journals(
+            name,
+            description,
+            public,
+            parent_id,
+            identifier,
+            active,
+            templated
+          )
+        SELECT name,
+          description,
+          public,
+          parent_id,
+          identifier,
+          active,
+          templated
+        FROM projects
+        RETURNING id,
+          identifier
+      ),
+      journals_insertion AS (
+        INSERT into journals (
+            journable_id,
+            journable_type,
+            user_id,
+            created_at,
+            version,
+            activity_type,
+            data_type,
+            data_id
+          )
+        SELECT projects.id,
+          'Project',
+          :user_id,
+          projects.created_at,
```

Comment:
I'd add the `update_at` column here as well and set it to the `projects.updated_at` since that is the last time the project was altered which would have been the time the journals would have been updated had the functionality existed back then. 

If we rely on the current timestamp, which is the default value, then it looks as if the change had been done when the migration ran e.g. in the roadmap.

Reply:
Nice catch :+1:

Reply:
1b2e668f974f6702e5958fb9b2f7188008e923a2